mobilerobot_joint_publisher:
  type: "joint_state_controller/JointStateController"
  publish_rate: 50

mobile_chassis_velocity_controller:
  type: "mecanum_wheel_controller/MecanumWheelController"
  front_left_wheel_joint: "forward_left_joint"
  back_left_wheel_joint: "backward_left_joint"
  front_right_wheel_joint: "forward_right_joint"
  back_right_wheel_joint: "backward_right_joint"
  publish_rate: 50
  pose_covariance_diagonal: [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 0.03]
  twist_covariance_diagonal: [0.001, 0.001, 0.001, 1000000.0, 1000000.0, 0.03]
  cmd_vel_timeout: 0.25

  # Override URDF look-up for wheel separation since the parent link is not the base_link.
  wheel_separation_x: 0.3
  wheel_separation_y: 0.3

  # Odometry fused with IMU is published by robot_localization, so
  # no need to publish a TF based on encoders alone.
  enable_odom_tf: true

  # Wheel separation and radius multipliers
  wheel_separation_multiplier: 1.0 # default: 1.0
  wheel_radius_multiplier    : 1.0 # default: 1.0

  mechanical_reduction: 156.0

  # Velocity and acceleration limits
  # Whenever a min_* is unspecified, default to -max_*
  linear:
    x:
      has_velocity_limits    : true
      max_velocity           : 0.35   # m/s
      has_acceleration_limits: false
      max_acceleration       : 1.0    # m/s^2
    y:
      has_velocity_limits    : true
      max_velocity           : 0.35   # m/s
      has_acceleration_limits: false
      max_acceleration       : 1.0    # m/s^2
  angular:
    z:
      has_velocity_limits    : true
      max_velocity           : 0.15   # rad/s
      has_acceleration_limits: false
      max_acceleration       : 1.0   # rad/s^2
